walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,134 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    public List<TimeRange> findRange( ArrayList timeRangesArray, MeetingRequest request, Collection<Event> event){
+        int duration = (int)request.getDuration();
+        List<TimeRange> rangezz = new ArrayList<TimeRange>();
+        int timeRange = 0;
+        int timePlus = 0;
+        for(int i = 0; i < timeRangesArray.size(); i++) {
+            timeRange = (int)timeRangesArray.get(i);
+            // if there are duplicates remove those instances 
+            if (i == 0) {  
+                // skip if curr time = Start of Day (0)
+                if (timeRange == TimeRange.START_OF_DAY) {
+                    continue;
+                }
+                rangezz.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, timeRange, false));
+            }
+            else if (i == (timeRangesArray.size() - 1)) {
+                // skip if curr time = End of Day (1440)
+                if (timeRange == (TimeRange.END_OF_DAY + 1)) {
+                    break;
+                }
+                rangezz.add(TimeRange.fromStartEnd(timeRange, TimeRange.END_OF_DAY, true));
+            }
+            else {
+                timePlus = (int)timeRangesArray.get(++i);
```

Comment:
Won't this double increment i? Shouldn't it be "i + 1" instead of "++i"? Or are you intentionally incrementing i a second time?

Reply:
I'm intentioally incrementing i a second time because I'm using both current index and the next index